Tasks/NuGetInstaller/vsonugethelper.ts

```diff
@@ -0,0 +1,184 @@
+
+        // load content of the user's nuget.config
+        var xml = fs.readFileSync(this._nugetConfigPath).toString();
+
+        // get package sources
+        xmlreader.read(xml, (err, res) => {
+            if (err) return tl.error(err);
+
+            var packageSources = [];
+            var packageSource: IPackageSource;
+            var sourceKey;
+            var sourceValue;
+            var promises: Q.Promise<{}>[] = [];
+            var deffered: Q.Deferred<{}>;
+            var deffers: Q.Deferred<{}>[] = [];
+
+            for (var i = 0; i < res.configuration.packageSources.add.count(); i++) {
+
+                sourceKey = res.configuration.packageSources.add.at(i).attributes().key;
+                sourceValue = res.configuration.packageSources.add.at(i).attributes().value;
+                packageSource = { id: i, feedName: sourceKey, feedUri: sourceValue, addCredential: false };
+                deffered = Q.defer();
+                deffers[i] = deffered;
+                promises.push(deffered.promise);
+
+                // check if need to add credential to feed
+                tl._writeLine('check credential: ' + sourceValue)
+                this.shouldGetCredentialsForFeed(packageSource).then((result) => {                         
+                    if (result.addCredential) {
+                        packageSources.push({
+                            key: result.feedName, 
+                            value: result.feedUri
+                        });
+                    }
+
+                    deffers[result.id].resolve(result);
+                });            
+            }
+
+            Q.allSettled(promises).then(() => {
+                // save nuget config file to agent build directory
+                tl._writeLine('save nuget.config to temp config file');
+                if (!(fs.existsSync(this.tempNugetConfigDir))) {
+                    fs.mkdirSync(this.tempNugetConfigDir);
+                };
+                fs.writeFileSync(this.tempNugetConfigPath, xml);
+                fs.chmodSync(this.tempNugetConfigPath, parseInt('0600', 8));
+         
+                // remove sources
+                tl._writeLine('remove sources in the config file');
+                this.removeSourcesInNugetConfig(packageSources);
+
+                // add sources
+                tl._writeLine('add sources in the config file');
+                this.addSourcesInNugetConfig(packageSources);
+
+                def.resolve(true);
+            },
+            (err) => {              
+                def.reject("failed to add sources");
+            });
+        });  
+
+        return def.promise;
+    }
+
+    private removeSourcesInNugetConfig(packageSources: any[]) {
+        packageSources.forEach((source) => {
+            var nugetTool = tl.createToolRunner(this._nugetPath);
+
+            nugetTool.arg('sources');
+            nugetTool.arg('remove');
+            nugetTool.arg('-name ' + source.key);
```

Comment:
You don't have to add the spaces in the arg '-name ' + source.key. Instead you can use nugetTool.arg('-name'); nugetTool.arg(source.key); That will add the spaces as you expect and I believe it is recommended.